src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputParameterConverter.cs

```diff
@@ -0,0 +1,123 @@
+                    || reader.TryReadWithConverter(nameof(InputParameter.VirtualParameter), options, ref virtualParameter)
+                    || reader.TryReadWithConverter(nameof(InputParameter.GroupedBy), options, ref groupBy)
+                    || reader.TryReadString(nameof(InputParameter.Kind), ref kind)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsRequired), ref isRequired)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsApiVersion), ref isApiVersion)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsResourceParameter), ref isResourceParameter)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsContentType), ref isContentType)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsEndpoint), ref isEndpoint)
+                    || reader.TryReadBoolean(nameof(InputParameter.SkipUrlEncoding), ref skipUrlEncoding)
+                    || reader.TryReadBoolean(nameof(InputParameter.Explode), ref explode)
+                    || reader.TryReadString(nameof(InputParameter.ArraySerializationDelimiter), ref arraySerializationDelimiter)
+                    || reader.TryReadString(nameof(InputParameter.HeaderCollectionPrefix), ref headerCollectionPrefix);
+
+                if (!isKnownProperty)
+                {
+                    reader.SkipProperty();
+                }
+            }
+
+            name = name ?? throw new JsonException("Parameter must have name");
+            nameInRequest = nameInRequest ?? throw new JsonException("Parameter must have nameInRequest");
+            parameterType = parameterType ?? throw new JsonException("Parameter must have type");
+
+            if (location == null)
+            {
+                throw new JsonException("Parameter must have location");
+            }
+            Enum.TryParse<RequestLocation>(location, ignoreCase: true, out var requestLocation);
+
+            if (kind == null)
+            {
+                throw new JsonException("Parameter must have kind");
+            }
+            Enum.TryParse<InputOperationParameterKind>(kind, ignoreCase: true, out var parameterKind);
+
+            var parameter = new InputParameter(name, nameInRequest, description, parameterType, requestLocation, defaultValue, virtualParameter, groupBy, parameterKind, isRequired, isApiVersion, isResourceParameter, isContentType, isEndpoint, skipUrlEncoding, explode, arraySerializationDelimiter, headerCollectionPrefix, GetSerializationFormat(parameterType, requestLocation));
+
+            if (id != null)
+            {
+                resolver.AddReference(id, parameter);
+            }
+
+            return parameter;
+        }
+
+        private static SerializationFormat GetSerializationFormat(InputType parameterType, RequestLocation requestLocation)
+        {
+            if (parameterType is InputPrimitiveType type && type.Kind == InputTypeKind.DateTime)
```

Comment:
maybe using pattern matching like:
```suggestion
            if (parameterType is InputPrimitiveType { Kind : InputTypeKind.DateTime})
```